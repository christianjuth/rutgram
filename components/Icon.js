import React from 'react';
import PropTypes from 'prop-types';
import { Image, TouchableOpacity, View, StyleSheet } from 'react-native';

function Icon(props) {

  let icons = {
    'person':   require('../assets/icons/person.png'),
    'person-o': require('../assets/icons/person-o.png'),
    'heart':   require('../assets/icons/heart.png'),
    'heart-o': require('../assets/icons/heart-o.png'),
    'heart-red':   require('../assets/icons/heart-red.png'),
    'add-box':   require('../assets/icons/add-box.png'),
    'add-box-o': require('../assets/icons/add-box-o.png'),
    'search':   require('../assets/icons/search.png'),
    'search-o': require('../assets/icons/search-o.png'),
    'home':   require('../assets/icons/home.png'),
    'home-o':   require('../assets/icons/home-o.png'),
    'message-o':   require('../assets/icons/message-o.png'),
  };

  // adjust to match expo icons
  let size = props.size + 2,
      height = size,
      width = size,
      Wrap = props.onPress ? TouchableOpacity : View;

  return(
    <Wrap
      style={[ styles.iconWrap, { height, width }, props.style ]}
      onPress={props.onPress}
      activeOpacity={0.7}
    >
      <Image resizeMode='contain' style={styles.icon} source={icons[props.name]} />
    </Wrap>
  );
}

Icon.propTypes = {
  name: PropTypes.string,
  onPress: PropTypes.func,
  style: PropTypes.object,
  size: PropTypes.number
};

const styles = StyleSheet.create({
  iconWrap: {
    alignItems: 'center',
    justifyContent: 'center'
  },

  icon: {
    flex: 1,
    height: '100%',
    width: '100%'
  }
});

export default Icon;
