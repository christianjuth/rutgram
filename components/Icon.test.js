import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Icon from './Icon';

configure({ adapter: new Adapter() });

describe('<DoubleTap />', () => {

  it('renders correctly', () => {
    const tree = renderer
      .create(<Icon name='person'/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('onPress fires correctly', () => {
    const mockFunc = jest.fn();
    const component = shallow(<Icon onPress={mockFunc} name='person'/>);
    component.simulate('press');
    expect(mockFunc).toHaveBeenCalledTimes(1);
  });
});
