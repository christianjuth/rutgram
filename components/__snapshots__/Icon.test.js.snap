// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DoubleTap /> renders correctly 1`] = `
<View
  activeOpacity={0.7}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "height": NaN,
        "width": NaN,
      },
      undefined,
    ]
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
